aws/resource_aws_route53_zone_association.go

```diff
@@ -163,24 +159,39 @@ func resourceAwsRoute53ZoneAssociationParseId(id string) (string, string, error)
 	return parts[0], parts[1], nil
 }
 
-func route53GetZoneAssociation(conn *route53.Route53, zoneID, vpcID string) (*route53.VPC, error) {
-	input := &route53.GetHostedZoneInput{
-		Id: aws.String(zoneID),
+func resourceAwsRoute53ZoneAssociationRefreshFunc(conn *route53.Route53, changeId, id string) resource.StateRefreshFunc {
+	return func() (interface{}, string, error) {
+		changeRequest := &route53.GetChangeInput{
+			Id: aws.String(changeId),
+		}
+		result, state, err := resourceAwsGoRoute53Wait(conn, changeRequest)
+		if isAWSErr(err, "AccessDenied", "") {
+			log.Printf("[WARN] AccessDenied when trying to get Route 53 change progress for %s - ignoring due to likely cross account issue", id)
+			return true, route53.ChangeStatusInsync, nil
+		}
+		return result, state, err
+	}
+}
+
+func route53GetZoneAssociation(conn *route53.Route53, zoneID, vpcID, vpcRegion string) (*route53.HostedZoneSummary, error) {
+	input := &route53.ListHostedZonesByVPCInput{
+		VPCId:     aws.String(vpcID),
+		VPCRegion: aws.String(vpcRegion),
 	}
 
-	output, err := conn.GetHostedZone(input)
```

Comment:
To prevent IAM permissions churn for operators with existing configurations, we should prefer to keep the old API call here if possible. I'm not sure adding the new `owning_account` attribute is worth the burden.

Reply:
Of course if this is required for the cross-account handling, then we'll just need to include a note in the CHANGELOG. 👍

Reply:
This is a required change for cross-account handling - the zone association is in the account holding the VPC, but you can't read the zone details from that account. ListHostedZonesByVPC does however work. I'll add a CHANGELOG line.

Reply:
I think @bflad normally does the CHANGELOG changes 😄 - here's the markdown ready:

```
NOTES:

* resource/aws_route53_zone_association: The addition of cross-account zone association support required the use of new `ListHostedZonesByVPC` API call. Restrictive IAM permissions for Terraform may require updates. [GH-14215]
```